### DISCUSSION TITLE
Fix x-persist garbage collection and 404 page "back" button styling issues creates a lot of bugs

### DISCUSSION THREAD

**Discussion by Graphicfort** [livewire/livewire#6823] [7. Bugs (archived)] [answered]
### Livewire version

3.0.4

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

<pre>
for (let child of Array.from(document.head.children)) {
    if (isAsset(child)) {
      if (!touchedHeadElements.some((i) => i.outerHTML === child.outerHTML)) {
        child.remove();
      }
    }
}
</pre>

1- This code is removing every style tag located at the head including nprogress styles, it means the progress will show up only once, because it's styles removed.
2- Calling any third party styles will be removed, I'm using overlayscrollbars and it's styled is removed

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Graphicfort oh that's not good! Any chance you can provide me a small example repo that demonstrates the issue, so I can investigate and rollback if needed? Thanks!

**Reply by Graphicfort** [reply]
Sorry I don't have but you can check this https://www.youtube.com/watch?v=Nv1jWLB5ZsQ
1- Reload the page and check for the style tags @ the head
2- Click on any navigate link
3- Check the style tags again and they will be gone

Please note that I'm using livewire as small components not a full page component

**Reply by joshhanley** [reply]
@Graphicfort thanks, I'll have a look at it

**Reply by joshhanley** [reply]
@Graphicfort just to confirm, is the layout on the first page (that include the styles) the same as the layout on the second page? Just trying to work out how I can replicate the issue

**Reply by Graphicfort** [reply]
No different views

**Reply by joshhanley** [reply]
@Graphicfort yeah the views would be different, but also different layouts? Like is the first page `app.blade.php` and the next `guest.blade.php` or are they both using `app.blade.php` to include the head tag and styles?

**Reply by Graphicfort** [reply]
Yes different, and using the controller to decide which on should be used. Also test it with the same layout/view but with different queries and I got the same result (the style tags removed)

**Reply by joshhanley** [reply]
@Graphicfort thanks! Hmm I can't replicate it, I have two different views with different layouts and the styles seems to be persisting/ updating/removing as required happily

**Reply by joshhanley** [reply]
@Graphicfort actually, got it, thanks! I should have read your description better than nprogress was being removed. Once I looked for that I noticed it happening. I should be able to write a test for that now.

**Reply by Graphicfort** [reply]
Nice, and to make it very simple with a simple code:

1- create a new/clean project

2- app.blade.php code
<pre>
&lt;html&gt;
&lt;head&gt;
    @livewireStyles
    @vite(['resources/js/app.js'])
&lt;/head&gt;
&lt;body&gt;
    &lt;nav&gt;
        &lt;a href="/" wire:navigate&gt;Dashboard&lt;/a&gt;
    &lt;/nav&gt;
    @livewireScriptConfig
&lt;/body&gt;
&lt;/html&gt;
</pre>

3- dashboard controller
<pre>
class Dashboard extends Controller
{
    //
    public function show() {
        return view( 'app' );
    }
}
</pre>

4- route:
<pre>
Route::get('/', [Dashboard::class, 'show']);
</pre>

and click on Dashboard link

**Reply by Graphicfort** [reply]
https://github.com/livewire/livewire/assets/13869984/80fa83fd-f221-4a29-aa9d-4b17b3496be7

**Comment by joshhanley** [comment]
@Graphicfort thanks for all your help! I've submitted a failing test PR #6824. I've asked Caleb for feedback, so will wait for that before either trying to fix or reverting.

**Comment by joshhanley** [comment, marked as answer]
@Graphicfort we reverted the troublesome code. This fix has been tagged and released in v3.0.5.
